Description: Update documentation

Reflect #767148 fix, high precision now being default, and releases.
Use the Debian documentation path in the error message.

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>

diff --git a/src/cl_device_id.c b/src/cl_device_id.c
index e9e2c16..3227340 100644
--- a/src/cl_device_id.c
+++ b/src/cl_device_id.c
@@ -610,7 +610,7 @@ cl_self_test(cl_device_id device, cl_self_test_res atomic_in_l3_flag)
                     } else {
                       ret = SELF_TEST_SLM_FAIL;
                       printf("Beignet: self-test failed: (3, 7, 5) + (5, 7, 3) returned (%i, %i, %i)\n"
-                             "See README.md or http://www.freedesktop.org/wiki/Software/Beignet/\n",
+                             "This can usually be fixed by upgrading Linux to >= 4.2,\nsee /usr/share/doc/beignet-opencl-icd/Beignet.mdwn.gz or http://www.freedesktop.org/wiki/Software/Beignet/\n",
                              test_data[0], test_data[1], test_data[2]);
 
                     }
diff --git a/docs/Beignet.mdwn b/docs/Beignet.mdwn
index 9a2b516..72cf25e 100644
--- a/docs/Beignet.mdwn
+++ b/docs/Beignet.mdwn
@@ -142,7 +142,7 @@ Supported Targets
 
  * 3rd Generation Intel Core Processors
  * Intel “Bay Trail” platforms with Intel HD Graphics
- * 4th Generation Intel Core Processors "Haswell", need kernel patch currently, see the "Known Issues" section.
+ * 4th Generation Intel Core Processors "Haswell", with some issues on Linux <= 4.1, see the "Known Issues" section.
  * 5th Generation Intel Core Processors "Broadwell".
 
 Known Issues
@@ -164,7 +164,7 @@ Known Issues
   forever until a reboot.
 
 * "Beignet: self-test failed" and almost all unit tests fail.
-  Linux 3.15 and 3.16 (commits [f0a346b](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=f0a346bdafaf6fc4a51df9ddf1548fd888f860d8)
+  Linux 3.15 and upstream 3.16 (_not_ Debian jessie 3.16; commits [f0a346b](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=f0a346bdafaf6fc4a51df9ddf1548fd888f860d8)
   to [c9224fa](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=c9224faa59c3071ecfa2d4b24592f4eb61e57069))
   enable the register whitelist by default but miss some registers needed
   for Beignet.
@@ -175,7 +175,7 @@ Known Issues
 
   On Haswell hardware, Beignet 1.0.1 to 1.0.3 also required the
   above workaround on later Linux versions, but this _should not_ be
-  required in current (after [83f8739](http://cgit.freedesktop.org/beignet/commit/?id=83f8739b6fc4893fac60145326052ccb5cf653dc))
+  required in Beignet 1.1 or current (after [83f8739](http://cgit.freedesktop.org/beignet/commit/?id=83f8739b6fc4893fac60145326052ccb5cf653dc))
   git master.
 
 * "Beignet: self-test failed" and 15-30 unit tests fail on 4th Generation (Haswell) hardware.
@@ -183,10 +183,11 @@ Known Issues
   Linux <= 4.0, and requires the i915.enable_ppgtt=2 [boot parameter](https://wiki.ubuntu.com/Kernel/KernelBootParameters)
   on Linux 4.1.
   
-  This will be fixed in Linux 4.2; older versions can be fixed with
-  [this patch](https://01.org/zh/beignet/downloads/linux-kernel-patch-hsw-support).
+  This is fixed from Linux 4.2, which is the default in sid/stretch, and available from -backports in jessie:
+  
+  `sudo apt-get install -t jessie-backports linux-image-amd64`
   
-  If you do not need \_\_local, you can override the self-test with
+  (or -i386).  If you do not need \_\_local, you can override the self-test with
   
   `export OCL_IGNORE_SELF_TEST=1`
   
diff --git a/docs/Beignet/Backend.mdwn b/docs/Beignet/Backend.mdwn
index 583e5d2..e0de7bf 100644
--- a/docs/Beignet/Backend.mdwn
+++ b/docs/Beignet/Backend.mdwn
@@ -37,9 +37,7 @@ Environment variables are used all over the code. Most important ones are:
   precision math instructions compliant with OpenCL Spec. So we provide a
   software version to meet the high precision requirement. Obviously the
   software version's performance is not as good as native version supported by
-  GEN hardware. What's more, most graphics application don't need this high
-  precision, so we choose 0 as the default value. So OpenCL apps do not suffer
-  the performance penalty for using high precision math functions.
+  GEN hardware.
 
 - `OCL_SIMD_WIDTH` `(8 or 16)`. Select the number of lanes per hardware thread,
   Normally, you don't need to set it, we will select suitable simd width for
